### PR TITLE
Hotfix: Resolve breaking change in PyJWT upgrade

### PR DESCRIPTION
**Pull request by jtwillis92**
### Summary of Changes
Addresses an issue introduced by [this Dependabot upgrade](https://github.com/raft-tech/TANF-app/pull/933)

The function signature for `jwt.decode` given by the PyJWT library changed in the upgrade such that the `algorithm` argument is now pluralized and accepts a list of strings instead of a single string. As of v2.0 this list must be explicitly supplied: [CHANGELOG](https://pyjwt.readthedocs.io/en/latest/changelog.html#require-explicit-algorithms-in-jwt-decode-by-default).

This was not caught during unit tests because all of the current calls to this function get mocked in our tests like so:
```
    mock_decode = mocker.patch("tdpservice.users.api.login.jwt.decode")
    decoded_token = {
        "email": "test@example.com",
        "email_verified": False,
        "nonce": nonce,
        "iss": "https://idp.int.identitysandbox.gov",
        "sub": "b2d2d115-1d7e-4579-b9d6-f8e84f4f56ca",
        "verified_at": 1577854800,
    }
    mock_post.return_value = MockRequest(data=token)
    mock_decode.return_value = decoded_token
```

Since this code was not executed during tests the error was never raised. Additionally, local testing requires a re-build of docker containers before this error will surface so it was further missed during that stage.

This PR resolves the login errors and adds a test to cover this function without being mocked so we can prevent this issue from occurring again if any future upgrades of the PyJWT alter the `decode` function.

### How to Test
* Pull this branch down locally
* Rebuild and restart the backend docker container:
```
docker-compose -f docker-compose.yml -f docker-compose.local.yml down
docker-compose -f docker-compose.yml -f docker-compose.local.yml build web
docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d web
```
* Navigate to the local frontend side and confirm authentication works as expected.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x] Tokens obtained by logging in to Login.gov can be successfully decoded by the TDRS backend
  - [x] Tests cover the `decode_payload` method of `TokenAuthorizationOIDC`

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

N/A no UX changes

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?
+ [x] If this PR introduces frontend code, is that code documented both inline and overall?
+ [x] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### CODE REVIEW

File: tdrs-backend/tdpservice/users/test/test_auth.py [hunk 1]
```diff
@@ -525,3 +526,21 @@ def test_generate_token_endpoint_parameters():
...
+
+
+def test_token_auth_decode_payload():
+    """Test ID token decoding functionality."""
+    decoded_token = TokenAuthorizationOIDC.decode_payload(
+        os.environ['MOCK_TOKEN'],
+        # Since these tokens are short lived our MOCK_TOKEN used for tests
+        # is expired and would need to be refreshed on each test run, to work
+        # around that we will disable signature verification for this test.
+        # TODO: Consider writing code to generate MOCK_TOKEN on demand
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/test/test_auth.py:538]: We should wait until we have a decision on NextGen XMS / ACF AMS before pursuing this TODO. If we choose to move away from login.gov most of this code will need to be rewritten.

File: tdrs-backend/tdpservice/users/api/login.py [hunk 2]
```diff
@@ -35,8 +35,12 @@ class InactiveUser(Exception):
 class TokenAuthorizationOIDC(ObtainAuthToken):
     """Define methods for handling login request from login.gov."""
 
-    def decode_payload(self, id_token):
+    @staticmethod
+    def decode_payload(id_token, options=None):
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/api/login.py:39]: The addition of the optional argument `options` was added to support disabling signature verification in tests. The default should be used in all cases other than testing.

**Review by jorgegonzalez — APPROVED** [review]
LGTM

**Review by abottoms-coder — APPROVED** [review]

**Comment by ADPennington** [comment]
@jtwillis92

-----
 test results from mac:
mac_test

----
test results from windows (after applying workaround prettier fix to spin up frontend:
`{"error":"Invalid Validation Code Or OpenID Connect Authenticator Down!"}`

**Review by ADPennington — APPROVED** [review]
### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  + [x] Tokens obtained by logging in to Login.gov can be successfully decoded by the TDRS backend
  + [x] Tests cover the decode_payload method of TokenAuthorizationOIDC
As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?

### Deliverable 4: Accessible

_n/a_

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?
### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_none detected_

File: tdrs-backend/tdpservice/users/api/login.py [hunk 3]
```diff
@@ -47,10 +51,10 @@ def decode_payload(self, id_token):
                 key=cert_str,
                 issuer=os.environ["OIDC_OP_ISSUER"],
                 audience=os.environ["CLIENT_ID"],
-                algorithm="RS256",
+                algorithms=["RS256"],
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/api/login.py:54]: For posterity, this argument was the breaking change. All other changes are to facilitate testing.

**Pull request merged**